apps/hash-graph/lib/authorization/src/backend.rs

```diff
@@ -0,0 +1,47 @@
+mod spicedb;
+
+use core::fmt;
+use std::error::Error;
+
+use error_stack::Report;
+
+pub use self::spicedb::{SpiceDb, SpiceDbConfig};
+use crate::zanzibar::{Affiliation, Consistency, Resource, StringTuple, Subject, Zookie};
+
+/// A backend for interacting with an authorization system based on the Zanzibar model.
+pub trait AuthorizationBackend {
+    /// Returns if the [`Subject`] has the specified permission or relation to an [`Resource`].
+    async fn check<S, A, R>(
+        &self,
+        subject: &S,
+        affiliation: &A,
+        resource: &R,
+        consistency: Consistency<'_>,
+    ) -> Result<CheckResponse, Report<CheckError>>
+    where
+        S: Subject + Sync,
+        A: Affiliation<R> + Sync,
+        R: Resource + Sync;
```

Comment:
Wait, isn't this automatically `!Send`? You can force that a trait implementation must be `Send` by using:

```suggestion
    fn check<S, A, R>(
        &self,
        subject: &S,
        affiliation: &A,
        resource: &R,
        consistency: Consistency<'_>,
    ) -> impl Future<Output = Result<CheckResponse, Report<CheckError>>> + Send + '_
    where
        S: Subject + Sync,
        A: Affiliation<R> + Sync,
        R: Resource + Sync;
```

Implementors can still use `async fn`, but are forced to be `Send`.

Reply:
The function is designed to not require `Send`, the values are never passed into an `await` block, only the reference.
This is the first time I use the `async trait` feature, if it turns out to be required I'll change it later